lib/content/message/predictions.ex

```diff
@@ -49,9 +50,55 @@ defmodule Content.Message.Predictions do
           platform: Content.platform() | nil,
           terminal?: boolean(),
           certainty: non_neg_integer() | nil,
-          crowding_data_confidence: :high | :low | nil
+          crowding_data_confidence: :high | :low | nil,
+          crowding_description: {atom(), atom()} | nil
         }
 
+  # Zone definitions based on number of emptier cars
+  # - Tuple represents a range of car numbers (inclusive)
+  # - List represents a specific set of car numbers that define the zone
+  # - If zone is nil, it's not relevant for that number of cars
+  @crowding_zone_definitions %{
+    1 => [
+      front: {1, 2},
+      back: {5, 6},
+      middle: {3, 4},
+      train_level: nil
+    ],
+    2 => [
+      front: {1, 3},
+      back: {4, 6},
+      middle: [[3, 4], [2, 4], [3, 5]],
+      train_level: nil
+    ],
+    3 => [
+      middle: {2, 5},
+      front: {1, 4},
+      back: {3, 6},
+      train_level: [[1, 3, 5], [2, 4, 6]]
+    ],
+    4 => [
+      middle: {2, 5},
+      front: {1, 5},
+      back: {2, 6},
+      train_level: [[1, 3, 4, 6], [1, 2, 4, 6], [1, 3, 5, 6]]
+    ],
+    5 => [
+      front: {1, 5},
+      back: {2, 6},
+      middle: nil,
+      train_level: [[1, 3, 4, 5, 6], [1, 2, 4, 5, 6], [1, 2, 3, 5, 6], [1, 2, 3, 4, 6]]
+    ],
+    6 => [
+      train_level: {1, 6}
+    ]
+  }
```

Comment:
Thinking out loud: This could be a place where pattern matching really shines. If we create a list that represents the status of each car, e.g. `[:f, :e, :e, :f, :f, :f]`, we could pattern match against that, along with the number of empty cars, and create a series of cases like: `{2, [_, _, _, :f, :f, :f]} -> :front`. That gives a nice visual way to express the logic, and would also allow us to fold in the special cases from the code below.

Reply:
Hmm yeah that's an interesting idea. I sort of like how the current approach condenses the logic but I can also see that it's a bit more obscure than expressing everything as patterns. Will give it a try and see how it turns out